plugins/easyimage/plugin.js

```diff
@@ -6,6 +6,141 @@
 ( function() {
 	'use strict';
 
+	var commandsToRefresh = [ 'easyimageFull', 'easyimageSide' ];
+
+	function isSideImage( widget ) {
+		return widget.element.hasClass( 'easyimage-side' );
+	}
+
+	function isFullImage( widget ) {
+		return !isSideImage( widget );
+	}
+
+	function refreshCommands( editor, commands ) {
+		CKEDITOR.tools.array.forEach( commands, function( command ) {
+			editor.getCommand( command ).refresh( editor );
+		} );
+	}
+
+	function createCommandRefresh( check ) {
+		return function( editor ) {
+			var widget = editor.widgets.focused;
+
+			if ( widget && widget.name === 'image' ) {
+				this.setState( ( check && check( widget ) ) ? CKEDITOR.TRISTATE_ON : CKEDITOR.TRISTATE_OFF );
+			} else {
+				this.setState( CKEDITOR.TRISTATE_DISABLED );
+			}
+		};
+	}
+
+	function createCommand( exec, refreshCheck ) {
+		return {
+			startDisabled: true,
+			contextSensitive: true,
+
+			exec: function( editor ) {
+				var widget = editor.widgets.focused;
+
+				exec( widget );
+
+				// We have to manually refresh commands as refresh seems
+				// to be executed prior to exec.
+				refreshCommands( editor, commandsToRefresh );
+			},
+
+			refresh: createCommandRefresh( refreshCheck )
+		};
+	}
+
+	var stylesLoaded = false;
+
 	CKEDITOR.plugins.add( 'easyimage', {
+		requires: 'image2,contextmenu,dialog',
+		lang: 'en',
+
+		onLoad: function() {
+			CKEDITOR.dialog.add( 'easyimageAlt', this.path + 'dialogs/easyimagealt.js' );
+		},
+
+		init: function( editor ) {
+			if ( !stylesLoaded ) {
+				CKEDITOR.document.appendStyleSheet( this.path + 'styles/easyimage.css' );
+				stylesLoaded = true;
+			}
+
+			if ( editor.addContentsCss ) {
+				editor.addContentsCss( this.path + 'styles/easyimage.css' );
+			}
+
+			editor.addCommand( 'easyimageFull', createCommand( function( widget ) {
```

Comment:
Can you extract command definition logic into a separate method? To improve `init()`'s readability? Also this function could encapsulate methods like `refreshCommands`, `createCommandRefresh`, `createCommand`.